### DISCUSSION TITLE
Tailwind CSS framework

### DISCUSSION THREAD

**Discussion by s314cy** [clicepfl/clic-website#234] [Ideas]
A CSS framework for easily and rapidly building both reactive and neat-looking CSS, without having to create numerous CSS classes within the stylesheet. Instead, directly incorporate prebuilt classes within HTML files. What do you think? https://tailwindcss.com/

My personal experience with the framework has been rather successful. It eases the process of CSS development a lot, especially when you're a non-CSS-expert like me.

**Comment by mochiruntime** [comment]
Yes, looks great! :+1: 

It's the eternal debate of whether CSS should describe entire UI components or only expose qualifiers that can be composed. There are many approaches (see BEM, OOCSS, SMACSS, ...), from my experience at the end of the day there is no right or wrong, and using a well-established framework will certainly help, especially at CLIC where the team and experiences will vary. However one may also consider that using a styling framework will make the website look less original (many other websites will use the same styling rules).

Anyway I think we should change the whole stack for many other reasons (see #66 for instance), I will post a new issue with some other considerations. I really want something that has a more rigorous approach to component-based UIs, while enforcing server-side rendering and progressive enhancement.
